move_base/src/move_base.cpp

```diff
@@ -560,6 +566,86 @@ namespace move_base {
     return global_pose;
   }
 
+  //publish feedback based on latest_plan.Called in context of executeCycle.
+  void MoveBase::publishFeedback(const geometry_msgs::PoseStamped& current_position)
+  {
+    //calculate distance to goal from current position
+    std::pair<bool, double> distance_info = calculateDistanceToGoal(current_position);
+    if(!distance_info.first)
+    {
+      return;
+    }
+    //calculate average velocity
+    ros::Time current_time = ros::Time::now();
+    double dist = distance_info.second;
+    calculateAverageVelocity(dist, current_time);
+
+    //publish feedback after X(30) iterations
+    if (prev_feedback_info_.average_velocity_iteration > 30) {
+      move_base_msgs::Feedback msg;
+      msg.dist_to_goal = dist;
+      msg.time_to_goal = prev_feedback_info_.calculateTimeToGoal(dist);
+      feedback_distance_pub_.publish(msg);
+    }
+
+    //save previous feedback info
+    prev_feedback_info_.prev_time = current_time;
+    prev_feedback_info_.dist_to_goal = dist;
+  }
+
+  //calculate distance from current position to the goal
+  std::pair<bool, double> MoveBase::calculateDistanceToGoal(const geometry_msgs::PoseStamped& current_position)
+  {
+    boost::unique_lock<boost::recursive_mutex> lock(planner_mutex_);
+    if(latest_plan_->empty())
+    {
+      return std::make_pair(false, 0.0);
+    }
+    std::vector<geometry_msgs::PoseStamped> plan;
+    plan = *latest_plan_;
```

Comment:
Seems like we need to pull from `controller_plan_` as they do some strange pointers swap here 
https://github.com/Blindnology/navigation/blob/6c672306073ebdd31d53d64b7a49322f0face013/move_base/src/move_base.cpp#L600
and here
https://github.com/Blindnology/navigation/blob/6c672306073ebdd31d53d64b7a49322f0face013/move_base/src/move_base.cpp#L847
So at the end, `controller_plan_` stores the last global plan, `latest_plan_` - the previous one, and `planner_plan_` - the one before the previous.
Please check if I'm not mistaken.

Reply:
Yes, the contoller_plan is the latest one, thanks